internals/plan/plan.go

```diff
@@ -974,13 +977,25 @@ func (layer *Layer) Validate() error {
 
 // Validate checks that the combined layers form a valid plan. See also
 // Layer.Validate, which checks that the individual layers are valid.
-func (p *Plan) Validate() error {
+func (p *Plan) Validate(workloads map[string]workload.Workload) error {
 	for name, service := range p.Services {
 		if service.Command == "" {
 			return &FormatError{
 				Message: fmt.Sprintf(`plan must define "command" for service %q`, name),
 			}
 		}
+		if service.Workload != "" {
+			if workloads == nil {
+				return &FormatError{
+					Message: fmt.Sprintf(`service %q cannot run in workload %q because workloads are not supported in %v`, name, service.Workload, cmd.DisplayName),
```

Comment:
Is this always an accurate message in this case? What if we just didn't specify any workload but are using a program that supports them?